mysite/blog/views.py

```diff
@@ -111,3 +113,27 @@ def send_mail(self, valid_data):
             subject=f"{valid_data['from_name']} recommends you read {self.post_object.title}",
             recipient_list=[valid_data["to_email"]],
         )
+
+
+def post_search(request):
+    form = SearchForm()
+    query = None
```

Comment:
If there is no query param named `query` is will be None if you use the dict.get()
```suggestion
    query = form.cleaned_data.get("query")
```